### PR TITLE
IDEX-2180: Git committer preferences

### PR DESCRIPTION
**Pull request by svor**
Add ability to configure git user name and email into the Preferences wizard.

@vparfonov

### CODE REVIEW

**Comment by skabashnyuk** [comment]
ok for me. @vparfonov  @sleshchenko  what about you?

**Comment by sleshchenko** [comment]
OK for me

File: plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/preference/CommitterPreferencePresenter.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                                        GitLocalizationConstant constant,
+                                        PreferencesManager preferencesManager) {
+        super(constant.committerTitle(), constant.committerPreferenceCategory(), null);
+        this.view = view;
+        this.preferencesManager = preferencesManager;
+
+        String userName = preferencesManager.getValue(COMMITTER_NAME);
+        String userEmail = preferencesManager.getValue(COMMITTER_EMAIL);
+
+        name = userName == null ? DEFAULT_COMMITTER_NAME : userName;
```
Comment (sleshchenko) [inline plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/preference/CommitterPreferencePresenter.java]: You can use Objects.firstNonNull from guava library

File: plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/preference/CommitterPreferencePresenter.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        preferencesManager.setValue(COMMITTER_NAME, name);
+        preferencesManager.setValue(COMMITTER_EMAIL, email);
+
+        dirty = false;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void revertChanges() {
+        view.setName(preferencesManager.getValue(COMMITTER_NAME));
```
Comment (sleshchenko) [inline plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/preference/CommitterPreferencePresenter.java]: What is happening when there is no committer name value in preferences?

**Pull request merged**